api/src/main/java/com/github/streamshub/console/api/security/OidcTenantConfigResolver.java

```diff
@@ -40,6 +63,44 @@ void initialize() {
         if (oidc.getIssuer() != null) {
             oidcConfig.getToken().setIssuer(oidc.getIssuer());
         }
+
+        getTlsConfiguration().map(TlsConfiguration::getTrustStore).ifPresentOrElse(
+            this::configureTruststore,
+            () -> logger.infof("No truststore configured for OIDC provider")
+        );
+    }
+
+    Optional<TlsConfiguration> getTlsConfiguration() {
+        String dotSeparatedSource = "oidc.provider.trust";
+        String dashSeparatedSource = "oidc-provider-trust";
+        return tlsRegistry.get(dotSeparatedSource).or(() -> tlsRegistry.get(dashSeparatedSource));
+    }
+
+    void configureTruststore(KeyStore truststore) {
+        File workDir = new File(workPath);
+        File truststoreFile;
+
+        try {
+            truststoreFile = File.createTempFile("oidc-provider-trust", "." + truststore.getType(), workDir);
```

Comment:
With a least privilege hat on, consider using the 3 arg form of File.createTempFile and pass attrs as
PosixFilePermissions.asFileAttribute(PosixFilePermissions.fromString("rw-------")

Reply:
Good idea, I'll add that.